docs/features.rst

```diff
@@ -965,6 +965,71 @@ that are embedded by default:
     $ curl -i http://example.com/people/?embedded={"author": 0}
     HTTP/1.1 200 OK
 
+Embedded callback Serialization
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+In order to ease the development of tree structure within Eve, new callback associated
+to the embedding solving of documents have been introduced.
+The callback gives the possibility to add layer of custom business logic, such
+as adding filter or modifying query as per the pre GET/ per fetch events, to the application.
+This callbacks, in particular, can ease the development of the following end-points:
+   1. carts/<cart_id> --> datasource: carts
+   2. carts/<cart_id>/products/<product_id> --> datasource: products
+   3. carts/<cart_id>/products/<product_id>/details/<detail_id> --> datasource: details
+
+Indeed, let's say that carts contains a reference to products and products to details
+within the documents and I want to allow the clients to request them embedded on each
+end-point.
+In this scenario there is nothing that oblige me to have unique id for product_id
+(as I might have the same product across several carts) and detail_id (as I might
+have some details across several product). Therefore, only the combination
+of all the field in the URLs above allow me to find one and only one document.
+Therefore, the consumer of the framework is available to use those callback in order
+to be able to customize this behaviour for his needs.
+
+Here is an example of how events are configured:
+
+.. code-block:: pycon
+
+   >>> def modify_query(field_name, subdocument, data_relation, subresources_query):
+   ...     subresources_query[field_name].update({"custom_id": subdocument["custom_id"]})
+
+   >>> app = Eve()
+   >>> app.on_embedding_resolving += modify_query
+
+You may use want to make the query result no results for a specific end-point:
```

Comment:
You probably mean "You may want to hook your callback to a specific resource endpoint instead. In the following example we are attaching a callback to the `products`  endpoint:"